File: typescript/pangram/pangram.ts
export default class Pangram {
  alphabet: {[letter: string]: boolean};
  input: string;

  constructor(input: string) {
    this.input = input.toLowerCase();
    this.alphabet = {'a': false, 'b': false, 'c': false, 'd': false, 'e': false, 'f': false, 'g': false, 'h': false,
    'i': false, 'j': false, 'k': false, 'l': false, 'm': false, 'n': false, 'o': false, 'p': false, 'q': false,
    'r': false, 's': false, 't': false, 'u': false, 'v': false, 'w': false, 'x': false, 'y': false, 'z': false};
  }

  isPangram(): boolean {
    if (this.input === '') {
      return false;
    }
    for (let i = 0; i < this.input.length; i++) {
      this.alphabet[this.input[i]] = true;
    }
    for(const letter in this.alphabet) {
      if(!this.alphabet[letter]) {
        return false;
      }
    }
    return true;
  }
}

File: kotlin/pangram/src/main/kotlin/Pangram.kt
class Pangram {
  companion object {
    fun isPangram(string: String): Boolean {
      val lowerCasedString = string.toLowerCase()
      // In order to solve this in linear time we need to precompute
      // a HashMap with all of the characters in the alphabet.
      val alphabetTracker: MutableMap<Char, Boolean> = CharRange('a', 'z').associate {
        Pair(it, false)
      }.toMutableMap()

      lowerCasedString.forEach {
        if(alphabetTracker.containsKey(it)) {
          alphabetTracker[it] = true
        }
      }
      return alphabetTracker.values.all { it }
    }
  }
}